tests/conftest.py

```diff
@@ -29,41 +29,43 @@ def node(host, request):
         'mimic': 13,
         'dev': 99
     }
-    if not request.node.get_marker(node_type) and not request.node.get_marker('all'):  # noqa E501
-        pytest.skip("Not a valid test for node type: %s" % node_type)
 
-    if request.node.get_marker("no_lvm_scenario") and lvm_scenario:
+    for marker in request.node.iter_markers():
+        if marker.name not in group_names and marker.name != 'all':
+            pytest.skip("%s: Not a valid test for node type: %s" % (request.function, group_names))
+
+    if request.node.get_closest_marker("no_lvm_scenario") and lvm_scenario:
         pytest.skip("Not a valid test for lvm scenarios")
 
-    if not lvm_scenario and request.node.get_marker("lvm_scenario"):
+    if not lvm_scenario and request.node.get_closest_marker("lvm_scenario"):
         pytest.skip("Not a valid test for non-lvm scenarios")
 
-    if request.node.get_marker("no_docker") and docker:
+    if request.node.get_closest_marker("no_docker") and docker:
         pytest.skip(
             "Not a valid test for containerized deployments or atomic hosts")
 
-    if request.node.get_marker("docker") and not docker:
+    if request.node.get_closest_marker("docker") and not docker:
         pytest.skip(
             "Not a valid test for non-containerized deployments or atomic hosts")  # noqa E501
 
-    if node_type == "mgrs" and ceph_stable_release == "jewel":
+    if group_names == "mgrs" and ceph_stable_release == "jewel":
```

Comment:
This needs to be ``if "mgrs" in group_names``. Same for ``nfss`` and ``iscsigws`` below.

Reply:
@andrewschoen updated, good catch, thanks